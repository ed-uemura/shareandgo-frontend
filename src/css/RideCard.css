.ride-card {
	display: flex;
	flex-flow: row nowrap;
	background-color: white;
	padding: 15px;
	align-items: center;
	justify-content: space-between;
	border-radius: 12px;
}

.driver-picture {
	display: flex;
	background-color: var(--SGGreen);
	width: 82px;
	height: 82px;
	border-radius: 100%;
	overflow: hidden;
	justify-content: center;
	margin: 0;
}

.driver-picture img {
	height: 100%;
}

.driver-info {
	margin: 0 10px;
}

.text-important {
	color: var(--SGGreen);
	font-size: var(--SGTextSize);
	font-weight: 700;
}

.ride-details {
	font-size: var(--SGTextDetails);
	color: var(--SGBlack);
}

.ride-info {
	text-align: right;
}

.star span {
	color: var(--SGGreen);
}

.star .material-symbols-outlined {
	font-size: 18px;
}

.star .filled {
	font-variation-settings: 'FILL' 1, 'wght' 400, 'GRAD' 0, 'opsz' 48;
}

@media screen and (max-width: 374px) {
	.driver-picture {
		width: 65px;
		height: 65px;
	}
}
