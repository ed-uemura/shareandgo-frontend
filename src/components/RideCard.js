import React from 'react';
import { Link } from 'react-router-dom';
import '../css/RideCard.css';

const RideCard = ({
	driverPhoto,
	driverName,
	pickupLocation,
	destination,
	date,
	time,
	price,
	link,
}) => {
	return (
		<Link to={link}>
			<div className='ride-card'>
				<div className='photo'>
					<figure className='driver-picture'>
						<img src={driverPhoto} alt={driverName} />
					</figure>
				</div>
				<div className='driver-info'>
					<p className='text-important'>{driverName}</p>

					<i className='star'>
						<span class='material-symbols-outlined filled'>star</span>
					</i>
					<i className='star'>
						<span class='material-symbols-outlined filled'>star</span>
					</i>
					<i className='star'>
						<span class='material-symbols-outlined filled'>star</span>
					</i>
					<i className='star'>
						<span class='material-symbols-outlined filled'>star</span>
					</i>
					<i className='star'>
						<span class='material-symbols-outlined'>star</span>
					</i>

					<p className='ride-details'>From: {pickupLocation}</p>
					<p className='ride-details'>To: {destination}</p>
				</div>
				<div className='ride-info'>
					<p className='ride-details'>{date}</p>
					<p className='ride-details'>{time}</p>
					<br></br>
					<p className='text-important'>{price}</p>
				</div>
			</div>
		</Link>
	);
};

export default RideCard;
