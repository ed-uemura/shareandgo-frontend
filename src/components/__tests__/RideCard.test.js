import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import RideCard from '../RideCard';
import DriverImage from '../../images/profile-picture-mockup.jpg';
import { MemoryRouter } from 'react-router-dom';


afterEach(() =>{
    cleanup();
});

test('matches snapshot', () =>{ 
    const testInfo ={
        driverPhoto : DriverImage,
        driverName : 'Mathew Pitts',
        driverRating : '3',
        pickupLocation : '700 Royal Street',
        destination: '31254 Charlotte Avenue',
        date: '10/15/2022',
        time: '9:45am',
        price: '$58.42',
        link: '/ride-info',
    }

    const tree = renderer.create(
        <MemoryRouter>
            <RideCard {...testInfo}/>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});