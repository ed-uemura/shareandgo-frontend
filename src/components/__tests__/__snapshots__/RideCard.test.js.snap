// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  href="/ride-info"
  onClick={[Function]}
>
  <div
    className="ride-card"
  >
    <div
      className="photo"
    >
      <figure
        className="driver-picture"
      >
        <img
          alt="Mathew Pitts"
          src="profile-picture-mockup.jpg"
        />
      </figure>
    </div>
    <div
      className="driver-info"
    >
      <p
        className="text-important"
      >
        Mathew Pitts
      </p>
      <div>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={0}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={1}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={2}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined"
            id={3}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined"
            id={4}
          >
            star
          </span>
        </i>
      </div>
      <p
        className="ride-details"
      >
        From: 
        700 Royal Street
      </p>
      <p
        className="ride-details"
      >
        To: 
        31254 Charlotte Avenue
      </p>
    </div>
    <div
      className="ride-info"
    >
      <p
        className="ride-details"
      >
        10/15/2022
      </p>
      <p
        className="ride-details"
      >
        9:45am
      </p>
      <br />
      <p
        className="text-important"
      >
        $58.42
      </p>
    </div>
  </div>
</a>
`;
